server/portal/apps/message/models.py

```diff
@@ -0,0 +1,85 @@
+import logging
+from django.db import models
+from django.conf import settings
+from django.utils import timezone
+
+logger = logging.getLogger(__name__)
+
+
+class IntroMessages(models.Model):
+    """IntroMessages
+
+    Used for storing the visited status of each of the Intro (formerly Welcome) messages.
+    """
+
+    user = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        related_name="+",
+        on_delete=models.CASCADE
+    )
+
+    datetime = models.DateTimeField(default=timezone.now, blank=True)
+
+    # Each variable represents that intro message status
+    # True means message has not been dismissed by user
+    component = models.CharField(max_length=300, default='')
+    unread = models.BooleanField(default=True)
+
+    # Make each type of IntroMessage unique
+    class Meta:
+        unique_together = ('user', 'component',)
+
+
+class CustomMessageTemplate(models.Model):
+    """CustomMessageTemplate
+
+    Used for storing admin-controlled messages for specific components that utilize CustomMessages.
+    """
+
+    MESSAGE_TYPES = [('info', 'Info'), ('success', 'Success'),
+                     ('warning', 'Warn'), ('error', 'Error')]
+
+    COMPONENTS = [('DASHBOARD', 'Dashboard'), ('DATA', 'Data Files'),
+                  ('APPLICATIONS', 'Applications'), ('ALLOCATIONS', 'Allocations'),
+                  ('HISTORY', 'History'), ('UI', 'UI'), ('ACCOUNT', 'Account')]
+
+    component = models.CharField(help_text='Component type', max_length=20, choices=COMPONENTS, default='Dashboard')
+    message_type = models.CharField(help_text='Message type', max_length=20, choices=MESSAGE_TYPES, default='info')
+    dismissible = models.BooleanField(default=True)
```

Comment:
> dismissible = models.BooleanField(default=True)

I wonder if `False` would be a better default - in the case of an admin adding a message and forgetting to mark it as non-dismissible, a user could dismiss the message before the admin fixed it as non-dismissible. What do you think?